packages/node_modules/@webex/plugin-meetings/src/locus-info/parser.js

```diff
@@ -1,118 +1,706 @@
   }
 
-  // THIS IS A NEW MEETING OBJECT
-  // // If meeting has no sequence, it means just initialized and will apply whatever in locus DTO
-  // if (isNewMeeting || !this.locus.sequence) {
-  //   meeting._updateWithDeltaLocus(deltaLocus, emitter);
-  //   meeting.locusDesync = false;
-  //   meeting.needToGetFullLocus = false;
-  //   return DELTAEVENT.GT;
-  // }
-
-  if (!deltaLocus.sequence) {
-    // TODO: force update
-  }
-
-  // TODO: If this was the first event, the client will not have a syncURL
-  // in its working copy to do a GET on. In this case clients should do a GET on locusUrl contained in the DTO
-
-  let res;
-  let resBase;
-
-  res = LocusUtil.compareLocus(deltaLocus.sequence, meeting.locusInfo.sequence);
-  // In-coming locus is greater than working copy, need to compare with base if there
-  // is base sequence except the first delta locus event
-  if (res === DELTAEVENT.GT && deltaLocus.baseSequence && meeting.locusInfo.syncUrl) {
-    resBase = LocusUtil.compareLocus(deltaLocus.baseSequence, meeting.locusInfo.sequence);
-    if (resBase === DELTAEVENT.LT || resBase === DELTAEVENT.CF) {
-      res = DELTAEVENT.CF;
-    }
-  }
-  // Special case for OBTP with meeting size equal or over 5. Locus might change this later
-  // This is a temporary fix. I will find a better solution later.
-  if (!deltaLocus.baseSequence && res === DELTAEVENT.EQ && deltaLocus.fullState.state === LOCUS.STATE.INITIALIZING &&
-        deltaLocus.sequence.rangeStart === 0 && deltaLocus.sequence.rangeEnd === 0 &&
-        deltaLocus.sequence.entries.length === 0) {
-    res = DELTAEVENT.GT;
-  }
-  switch (res) {
-    case DELTAEVENT.LT:
-    case DELTAEVENT.EQ:
-      meeting.locusDesync = false;
-      meeting.needToGetFullLocus = false;
-      break;
-    case DELTAEVENT.CF:
-      LocusUtil.generateSyncDebugFlag(meeting, deltaLocus);
-      if (meeting.locusDesync) {
-        meeting.meetingRequest.getFullLocus({
-          desync: true,
-          locusUrl: meeting.locusUrl
-        }).then((res) => {
-          meeting.locusInfo.onFullLocus(res.body);
-        });
+
+  /**
+   * Checks if two sequences overlap in time,
+   * the sequence with the higher minimum value is greater.
+   * Chooses sequence with most recent data.
+   * @param {Types~Locus} current
+   * @param {Types~Locus} incoming
+   * @returns {string} loci comparison state
+   */
+  static checkSequenceOverlap(current, incoming) {
+    let comparison = null;
+
+    // if earliest working copy sequence is more recent than last incoming sequence
+    if (current.min > incoming.max) {
+      // choose left side (current)
+      comparison = `${Parser.loci.GT}:SO001`;
+    }
+    // if last working copy sequence is before the earliest incoming sequence
+    else if (current.max < incoming.min) {
+      // choose right side (incoming)
+      comparison = `${Parser.loci.LT}:SO002`;
+    }
+
+    // if no match above, defaults to null
+    return comparison;
```

Comment:
Shouldn't this be `Parser.loci.EQ`?

Reply:
This follows the Loci Comparison Sequence algorithm and returning EQ here would produce unexpected results. The intent is to return `null` so that the next check can be applied.

Reply:
I know it seems like it should be EQ but just because the two conditions above don't apply, according to https://sqbu-github.cisco.com/WebExSquared/cloud-apps/wiki/Locus-Sequence-Comparison-Algorithm it doesn't necessarily mean they are equal. Other checks need to be applied. Returning `null` allows the other checks to continue.